spot-contracts/contracts/strategies/FeeStrategy.sol

```diff
@@ -0,0 +1,226 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.19;
+
+import { IFeeStrategy, IERC20Upgradeable } from "../_interfaces/IFeeStrategy.sol";
+import { IPerpetualTranche, IBondController } from "../_interfaces/IPerpetualTranche.sol";
+import { IVault } from "../_interfaces/IVault.sol";
+
+import { MathUpgradeable } from "@openzeppelin/contracts-upgradeable/utils/math/MathUpgradeable.sol";
+import { SafeCastUpgradeable } from "@openzeppelin/contracts-upgradeable/utils/math/SafeCastUpgradeable.sol";
+import { AddressUpgradeable } from "@openzeppelin/contracts-upgradeable/utils/AddressUpgradeable.sol";
+import { OwnableUpgradeable } from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import { Sigmoid } from "../_utils/Sigmoid.sol";
+import { BondHelpers } from "../_utils/BondHelpers.sol";
+import { PerpHelpers } from "../_utils/PerpHelpers.sol";
+
+/**
+ *  @title FeeStrategy
+ *
+ *  @notice This contract determins perp's mint, burn and rollover fees.
+ *
+ *          Fees are computed based on the deviationFactor, ie the ratio between
+ *          the current TVL to the expcted TVL in the vault system.
+ *
+ *              expctedTVL = perpTVL / trancheRatio
+ *              deviationFactor = currentTVL/expectedTVL
+ *
+ *          1) The mint fees are turned on when deviationFactor < 1,
+ *             and is a fixed percentage fee set by the owner.
+ *
+ *          2) The burn fees are turned on when deviationFactor > 1,
+ *             and is a fixed percentage fee set by the owner.
+ *
+ *          3) The rollover fees are signed and can flow in either direction.
+ *             The fee is a percentage of the tranches rolled over and is computed
+ *             through a sigmoid function. The slope and asymptotes are set by the owner.
+ *
+ *              rotationsPerYear = mintingBondDuration / 1_year
```

Comment:
However, `maxMaturityDuration` can be updated anytime by the owner. Its safer to rely on a bond which is already in the system (ie the deposit bond).

Reply:
Good point. At least this is bounded between `maxMaturityDuration` and `maxMaturityDuration - issueWindowOffset`. Since the issuer at the perp level, and the issuer params at the issuer level are both configurable though, in practice we've just updated the issuer – we may want to consider making the issuer params  immutable in a future update